src/io.jl

```diff
@@ -0,0 +1,90 @@
+"""
+    GeoJSON.read(json::String; lazyfc=false, ndim=2)
+
+Read GeoJSON from a string or file to a GeoInterface.jl compatible object.
+Set `ndim=3` for 3D geometries, which is also tried automatically when
+
+When reading in huge featurecollections (1M+ features), set `lazyfc=true`
+to only parse them into memory when accessed.
+"""
+function read(io; lazyfc=false, ndim=2, numbertype=Float32)
```

Comment:
Shouldn't we default to numbertype Float64? The numbertype is missing from the docstring.
Also perhaps we should rename `lazyfc` to `lazy`, and just document that it only works for FeatureCollections.

Reply:
I see you commented on the reason for Float32 [here](https://github.com/JuliaGeo/GeoJSON.jl/pull/63#issuecomment-1400133216). The number of decimals to store in the GeoJSON is discussed here: https://www.rfc-editor.org/rfc/rfc7946#section-11.2. I would lean to using a safe default of Float64, but am fine with Float32 as well.

Reply:
https://xkcd.com/2170/

Reply:
Hehe yeah I'm aware that for lat/lon it doesn't make much sense. I'm thinking more about projected coordinates, which can also be used, the spec mentions:

> However, where all involved parties have a prior arrangement, alternative coordinate reference systems can be used without risk of data being misinterpreted.

But even then a typical dutch y coordinate gives 3cm precision (`eps(400_000.0f0) => 0.03125f0`), which for most usecases should be fine. And if it's not it's easy to request more precision with `numbertype`.